### PR TITLE
Feature/uot 105734 - Evaluation Pipeline

### PR DESCRIPTION
**Pull request by katerdowdy**
For testing the retriever, similarity model, and QA model with their original fine-tuning data and in-domain GC data. 

New files:
- README.md - added stuff about how to use the classes/run all of them to test
- `scripts/run_evaluation.py` - for testing all the model pipelines at once
- `src/model_testing/evaluation.py` - has the pipeline classes (different subclass for each validation dataset). So for testing just the encoder model after we make a new sentence index, we would use the `IndomainRetrieverEvaluator` to check if the gold standard docs are still showing up in the new index.
- `src/model_testing/validation.py` - has classes for each of the validation datasets to get them in a format to use for testing. For QA and the Retriever in domain datasets, we're also checking that the answers are in the context/docs in the index before we include those in the evaluation.
- `src/utilities/model_helper.py` - we could move these, functions for opening and saving files mostly
- `data/validation/QA_domain_data.json` - we might want to change this a little to make our QA data more versatile (we can use it for evaluating the retriever and sim model) but showing the format of the examples we have now. 

Old files:
- `configs/config.py` - added more configs for the other transformers and validation datasets
- `src/search/sent_transformer/model.py` - I had to change some of this to be able to test the retriever and sim model separately, but I was able to use it to make an index locally so I think it should still work.
- `src/search/QA/QAReader.py` - preserving caps in questions for better answers

### CODE REVIEW

File: gamechangerml/configs/config.py [hunk 1]
```diff
@@ -2,6 +2,9 @@
 from os import environ
 import os
 from gamechangerml import REPO_PATH
+#from gamechangerml.api.utils.pathselect import get_model_paths
```
Comment (cWren0110) [inline gamechangerml/configs/config.py:5]: Should these commented lines be removed?

File: gamechangerml/scripts/run_evaluation.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from gamechangerml.src.model_testing.evaluation import SQuADQAEvaluator, IndomainQAEvaluator, IndomainRetrieverEvaluator, MSMarcoRetrieverEvaluator, SimilarityEvaluator
```
Comment (cWren0110) [inline gamechangerml/scripts/run_evaluation.py:1]: Are the metrics generated in this script captured anywhere? Confluence, MLFlow, etc.?

Comment (rha930) [reply gamechangerml/scripts/run_evaluation.py:1]: they will be soon, i think originally they were on confluence from prince for baseline stuff

Comment (katerdowdy) [reply gamechangerml/scripts/run_evaluation.py:1]: The metrics are explained somewhat in the README in model_testing, but I also updated Confluence here: https://confluence.di2e.net/x/jDpPT. If anybody has thoughts on other metrics we want, we can add those

**Review by cWren0110 — APPROVED** [review]

**Comment by rha930** [comment]
is the old stuff OBE in model_test/

**Comment by katerdowdy** [comment]
> is the old stuff OBE in model_test/

I just removed the old files

**Review by rha930 — CHANGES_REQUESTED** [review]
QA in the API params need to be fixed

**Comment by katerdowdy** [comment]
> QA in the API params need to be fixed

This should be working now, here's what I get from health check in API: ```gc-ml-gpu             | 2021-08-02 19:05:41,982 [INFO][PID:18]: -- Sentence index name: gamechangerml/models/sent_index
gc-ml-gpu             | 2021-08-02 19:05:41,983 [INFO][PID:18]: -- QE model name: gamechangerml/models/qexp_20201217
gc-ml-gpu             | 2021-08-02 19:05:41,984 [INFO][PID:18]: -- QA model name: gamechangerml/models/transformers/bert-base-cased-squad2
gc-ml-gpu             | 2021-08-02 19:07:41,899 [INFO][PID:18]: API Health Check
gc-ml-gpu             | 2021-08-02 19:07:41,902 [INFO][PID:18]: Model Health: GOOD
gc-ml-gpu             | 2021-08-02 19:07:41,903 [INFO][PID:18]: -- Sentence Transformer model name: {'encoder': 'gamechangerml/models/transformers/msmarco-distilbert-base-v2', 'sim': 'gamechangerml/models/transformers/distilbart-mnli-12-3'}```

**Review by rha930 — APPROVED** [review]

**Pull request merged**